src/components/builds/ConfigurationWithIssues.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { createStyles, Theme } from '@material-ui/core';
+import { WithStyles, withStyles } from '@material-ui/core/styles';
+import { Alert } from '@material-ui/lab';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    configurationTable: {
+      color: theme.palette.type === 'dark' ? theme.palette.primary.light : theme.palette.primary.dark,
+      background: theme.palette.type === 'dark' ? theme.palette.primary.dark : theme.palette.primary.light,
+      width: '100%',
+      borderSpacing: '0px',
+    },
+    lineNumber: {
+      // An attempt to be consistent in colors with chips[1] backgrounds used on the same builds page
+      //
+      // [1]: https://github.com/mui-org/material-ui/blob/bda3d8541b2f3bea14ed241ae8155239d524a24c/packages/material-ui/src/Chip/Chip.js#L14
+      background: theme.palette.type === 'dark' ? theme.palette.grey[700] : theme.palette.grey[300],
+      fontFamily: 'Monaco, monospace',
+      paddingRight: '5px',
+      width: '1%',
+      minWidth: '40px',
+      textAlign: 'right',
+    },
+    lineContent: {
+      fontFamily: 'Monaco, monospace',
+      paddingLeft: '5px',
+      whiteSpace: 'pre-wrap',
+    },
+    issue: {
+      fontSize: theme.typography.fontSize,
+    },
+  });
+
+interface Issue {
+  level: string;
+  message: string;
+  line: number;
+  column: number;
+}
+
+interface Props extends WithStyles<typeof styles> {
+  configuration: string;
+  issues: ReadonlyArray<Issue>;
+}
+
+function ConfigurationWithIssues(props: Props) {
+  let { configuration, issues, classes } = props;
+
+  // Sort issues by (line, column) and store them in a map for faster access
+  var sortedIssues = issues.slice().sort(function (left, right) {
+    return left.line - right.line || left.column - right.column;
+  });
+
+  let issueMap = {};
+
+  for (let issue of sortedIssues) {
+    let targetList = issueMap[issue.line] || (issueMap[issue.line] = []);
+    targetList.push(issue);
+  }
+
+  // Render configuration with issues
+  function getIssuesForLine(lineNumber: number) {
+    const issuesForLine = issueMap[lineNumber];
+
+    if (issuesForLine === undefined || issuesForLine.length === 0) return null;
+
+    const renderedIssues = issuesForLine.map(issue => {
+      return (
+        <Alert className={classes.issue} severity={issue.level.toLowerCase()}>
+          {issue.message}
+        </Alert>
+      );
+    });
+
+    return (
+      <tr>
+        <td colSpan={2}>{renderedIssues}</td>
```

Comment:
There is a weird padding here:

<img width="226" alt="Screen Shot 2021-03-25 at 10 18 24 AM" src="https://user-images.githubusercontent.com/989066/112490662-f493de00-8d55-11eb-954b-2fd0f42f0cec.png">

You can probably add the following class

```js
issue: {
  fontSize: theme.typography.fontSize,
}
```

And then
```suggestion
        <td className={classes.issues} colSpan={2}>{renderedIssues}</td>
```

Reply:
Not sure what this is supposed to accomplish (the class you've proposed to add already exists, for example), but I've fixed the padding here: https://github.com/cirruslabs/cirrus-ci-web/pull/351/commits/6c92c68f681d9426f26a4a212056a6f10a3de52f.

Looks like this now:

![Screenshot 2021-03-25 at 18 23 30](https://user-images.githubusercontent.com/85709/112498067-3bee8e80-8d97-11eb-85c7-e5e0fbb2832d.png)

Reply:
My bad. I meant a new class

```js
issues: {
  padding: 0,
},
```

`cellPadding` also works